packages/modals/src/elements/DrawerModal/DrawerModal.tsx

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+/* eslint-disable */
+
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useContext,
+  useCallback,
+  createRef,
+  forwardRef,
+  RefAttributes,
+  HTMLAttributes,
+  PropsWithoutRef,
+  ForwardRefExoticComponent
+} from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import mergeRefs from 'react-merge-refs';
+import { ThemeProps, DefaultTheme, ThemeContext } from 'styled-components';
+import { useModal } from '@zendeskgarden/container-modal';
+import { useDocument } from '@zendeskgarden/react-theming';
+import { useFocusVisible } from '@zendeskgarden/container-focusvisible';
+import { ModalsContext } from '../../utils/useModalContext';
+import {
+  StyledDrawerModalBackdrop,
+  StyledDrawerModal,
+  StyledDrawerModalFooter,
+  StyledDrawerModalFooterItem
+} from '../../styled';
+
+import { Header } from './Header';
+import { Body } from './Body';
+import { Close } from './Close';
+
+interface IStaticDrawerModalExport<T, P>
+  extends ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>> {
+  Header: typeof Header;
+  Body: typeof Body;
+  Close: typeof Close;
+  Footer: typeof StyledDrawerModalFooter;
+  FooterItem: typeof StyledDrawerModalFooterItem;
+}
+
+export interface IDrawerModalProps {
+  /**
+   * Determines whether the `DrawerModal` is opened or closed
+   */
+  isOpen?: boolean;
+  /**
+   * HTML attributes to spread onto backdrop element
+   */
+  backdropProps?: any;
+  /**
+   * Callback when a close action has been completed.
+   * Can be triggered from the backdrop and Close icon.
+   * @param {Object} event - DOM event that triggered the close action
+   */
+  onClose?: (event: TransitionEvent) => void;
+  /**
+   * Enable drawer modal animation
+   */
+  isAnimated?: boolean;
```

Comment:
`isAnimated={false}` does not appear to be working as expected.

Reply:
Whoops, yeah this prop isn't passed through to `StyledBackdrop`. 

I'm not sure if there is a non-transitioned `DrawerModal` (e.g transitioning the `transform`). @rossmoody should the `DrawerModal` have an option to disable the "slide-in" transition? If so, I could bake the ability to enable/disable transition into the `isAnimated` prop.

Reply:
I would be curious the rationale, but considering this functionality exists on all other components, I think it's safe to assume we'd want it here too.

Reply:
The `Modal` animates the backdrop background color. `DrawerModal` does not have a colored backdrop to animate or transition. But other components such as `Accordion` and `Steppers` **don't** have the ability to turn off transitions.

Reply:
Nice. Do we know why it was included initially?

Reply:
The original `isAnimated` prop also handles some entrance transitions for `Modal` I believe. We've had some areas disable animation from designer guidance, but I don't remember a rhyme or reason behind it.

In general, having the option to reduce animation globally is an [important a11y characteristic](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion). Normally this is handled by a global media query, but doing it at the component level could be helpful 🤷

Reply:
We've also seen iframe contexts get wonky out of Garden's control and turn off animation. So we provide the prop, as a rule.

Reply:
One more: the ability to knock-out animation is another feature of Garden that reinforces an open-source concept, allowing users to make it theirs by applying their own animation rubric (assuming they can hang with our sizes/shapes).